### PR TITLE
[DISCO-2487] - mdbook setup

### PR DESCRIPTION
**Pull request by tiftran**
## References

JIRA: [DISCO-2487](https://mozilla-hub.atlassian.net/browse/DISCO-2487)

## Description
Initial mdbook setup. Haven't looked into the existing docs, will do so in the follow issues.

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [X] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/contile/blob/main/CONTRIBUTING.md)
- [X] This PR conforms to the [Opsec policies](https://github.com/mozilla-services/websec-check)
- [ ] [Functional and performance test](https://github.com/mozilla-services/contile/tree/main/test-engineering) coverage has been expanded and maintained (if applicable)
- [ ] The `[load test: (abort|warn)]` keywords are applied (if applicable)
- [ ] Documentation has been updated
- [X] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title

[DISCO-2487]: https://mozilla-hub.atlassian.net/browse/DISCO-2487?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by mtbottle — COMMENTED** [review]
Can we update the "In-repo documentation" link on the[ main README ](https://github.com/mozilla-services/contile/blob/main/README.md)to link to the newly created `mdbook` page: https://mozilla-services.github.io/contile/?

File: .circleci/config.yml [hunk 1]
```diff
@@ -342,7 +342,58 @@ jobs:
...
+          command: touch workspace/doc/.nojekyll
+      - run:
+          name: Set remote origin if needed
+          command: |
+            git remote add origin git@github.com:mozilla-services/contile.git || true
+      - run:
+          name: Deploy docs to gh-pages
+          command: |
+            npx --yes gh-pages@3.0.0 \
+              --user "ci-build <ci-build@merino.mozilla.org>" \
```
Comment (ncloudioj) [inline .circleci/config.yml:393]:
```suggestion
              --user "ci-build <ci-build@contile.mozilla.org>" \
```

**Review by ncloudioj — APPROVED** [review]
LGTM, thanks for setting this up!

**Review by mtbottle — APPROVED** [review]
🚀

**Review by ncloudioj — APPROVED** [review]
Oops, sorry for the delay.

**Pull request merged**